baybe/acquisition/base.py

```diff
@@ -24,15 +28,37 @@ class AcquisitionFunction(ABC, SerialMixin):
     _abbreviation: ClassVar[str]
     """An alternative name for type resolution."""
 
-    def to_botorch(self, surrogate: Surrogate, best_f: float):
+    @classproperty
+    def is_mc(cls) -> bool:
+        """Flag indicating whether this is a Monte-Carlo acquisition function."""
+        return cls._abbreviation.startswith("q")
+
+    def to_botorch(
+        self,
+        surrogate: Surrogate,
+        train_x: pd.DataFrame,
```

Comment:
I know that **currently** the caller (i.e. `BayesianRecommender`) will always provide training data, so having it here as mandatory argument works. However, this will be relaxed once we get the #130 running. Therefore: can we make this optional here as well? I think that's should be enabled ...

Reply:
not sure if I get your connection to the other PR
but in that sense then `train_y` should also be optional as not eery acqf needs it?

Reply:
Sure, this covers both `train_x` and `train_y` (and reminder, after #220 this will be merged into one dataframe). 

The connection to the other PR: Currently, this part blocks `None`-valued variables to pass further down into the the `to_botorch` call:
<img width="685" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/a6f68556-6587-4400-a211-4dbb72bc40d9">
However, that code piece will be removed in #130 since execution with empty/non-existing dataframes will then be allowed. And at that point, your code would break. So either we fix your code (which I think is the preferred solution), or we need to replace the check with something else, i.e. conversion to empty dataframes (which is rather inelegant as it will still cause downstream problems in your method, e.g. for the `train_y.max().item()`)

Reply:
oki they are optional now

Reply:
I think that's only half the story unfortunately. The tricky part is to actually handle the case where `None` is passed, because your stuff will break (and I suspect, the pipeline run should fail because now you'll get `mypy` issues!?)

Reply:
indeed, prediction was correct 😶

Reply:
ok getting mypy errors now (not really surprising)
if you want me to make them optional as part of this PR we also have to include their logic of how to compute the derived quantities (best_f and X_baseline) which i suggest you do in a thought through manner in the PR that is supposed to enable these trainless predictions, which is not this PR